### PR TITLE
fix: pool fee event dispatch order

### PR DESCRIPTION
**Pull request by wischli**
# Description

* Context: https://centrifugedao.slack.com/archives/C04R42WGM7U/p1710854795914679
* Changes event order dispatch of pool creation and pool fees added when creating a new pool: Fee event needs to be after pool creation one

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by lemunozm — APPROVED** [review]
Clean and with its own test. Perfect!!

**Review by hieronx — COMMENTED** [review]
Changes re the pool fee event look good!

Something I realized that is completely unrelated to this PR, but was looking at the `pool-registry` code: shouldn't we emit an `MetadataSet` event here: https://github.com/centrifuge/centrifuge-chain/blob/9618071df487ebdbad1aafe17605b77f91a85b60/pallets/pool-registry/src/lib.rs#L289 Now we are only emitting it on updates,  which makes tracking all metadata harder as you need separate logic to track the first one.

If we do that, we should emit this event after the `T::ModifyPool::create` call

**Review by lemunozm — DISMISSED** [review]
LGTM!!

File: pallets/pool-system/src/impls.rs [hunk 1]
```diff
@@ -154,24 +154,6 @@ impl<T: Config> PoolMutate<T::AccountId, T::PoolId> for Pallet<T> {
...
-			// currency. Otherwise, we'd always need to convert investments to the decimals
-			// of tranche tokens and vice versa
-			let decimals = match T::AssetRegistry::metadata(&currency) {
-				Some(metadata) => metadata.decimals,
-				None => return Err(Error::<T>::MetadataForCurrencyNotFound.into()),
-			};
-
-			let metadata = tranche.create_asset_metadata(decimals, token_name, token_symbol);
-
-			T::AssetRegistry::register_asset(Some(tranche.currency.into()), metadata)
```
Comment (lemunozm) [inline pallets/pool-system/src/impls.rs:171]: Why is it no longer registered?

Comment (wischli) [reply pallets/pool-system/src/impls.rs:171]: It is! This was just moved down a few lines (L208+).

Comment (wischli) [reply pallets/pool-system/src/impls.rs:171]: The reason why we want to move this closure is to change the dispatch order of the events: Pool creation should be first and then the tranche token registration events.

**Review by lemunozm — APPROVED** [review]

**Pull request merged**